### PR TITLE
Dlq introduced

### PR DESCRIPTION
**Pull request by filiphaftek**
DLQ mechanism for forwarding errors. AWS has retry on every delivery, in case of fail a message is moved to dead-letter-exchange and then put on dead-letter-queue.

### CODE REVIEW

File: rabbitmq/consumer.go [hunk 1]
```diff
@@ -67,59 +67,45 @@ func (c Consumer) Start(forwarder forwarder.Client, check chan bool, stop chan b
 }
 
 func (c Consumer) connect() (<-chan amqp.Delivery, *amqp.Connection, *amqp.Channel, error) {
+	deadLetterExchangeName := c.ExchangeName + "-dead-letter"
```
Comment (jakub-tokaj-ah) [inline rabbitmq/consumer.go]: dead letter exchange has nothing to do with original exchange, it's rather connected to the queue and particular consumer - think what would happen when we created more than one queue with dead letter this way and one of the consumers worked while the other one caused errorors :)

File: rabbitmq/consumer.go [hunk 2]
```diff
@@ -67,59 +67,45 @@ func (c Consumer) Start(forwarder forwarder.Client, check chan bool, stop chan b
...
-		"topic",        // type
-		true,           // durable
-		false,          // auto-deleted
-		false,          // internal
-		false,          // no-wait
-		nil,            // arguments
-	)
-	if err != nil {
-		return failOnError(err, "Failed to declare an exchange")
+	// regural exchange
```
Comment (jakub-tokaj-ah) [inline rabbitmq/consumer.go]: typo

File: rabbitmq/consumer.go [hunk 3]
```diff
@@ -67,59 +67,45 @@ func (c Consumer) Start(forwarder forwarder.Client, check chan bool, stop chan b
 }
 
 func (c Consumer) connect() (<-chan amqp.Delivery, *amqp.Connection, *amqp.Channel, error) {
```
Comment (jakub-tokaj-ah) [inline rabbitmq/consumer.go]: i'm not sure if connect is the right name, it does way more than that, maybe `setup` or `init`

File: rabbitmq/consumer.go [hunk 4]
```diff
@@ -67,59 +67,45 @@ func (c Consumer) Start(forwarder forwarder.Client, check chan bool, stop chan b
...
-		c.ExchangeName, // exchange
-		false,
-		nil)
-	if err != nil {
-		return failOnError(err, "Failed to bind a queue")
+	// dead-letter-queue
+	if _, err = ch.QueueDeclare(deadLetterQueueName, true, false, false, false, nil); err != nil {
+		return failOnError(err, "Failed to declare a queue:"+deadLetterQueueName)
+	}
+	if err = ch.QueueBind(deadLetterQueueName, "#", deadLetterExchangeName, false, nil); err != nil {
```
Comment (jakub-tokaj-ah) [inline rabbitmq/consumer.go]: i think that this method does too much, some logic can be extracted (initialization of exchanges/queues/bindings for example)

File: rabbitmq/consumer.go (hunk 1 above)
Comment (filiphaftek) [inline rabbitmq/consumer.go]: Good spot, thx.

File: rabbitmq/consumer.go (hunk 3 above)
Comment (filiphaftek) [inline rabbitmq/consumer.go]: I'll split it into separate methods.

**Comment by filiphaftek** [comment]
@jakub-tokaj-ah conusmer name is defined as private as there is a method Name() in Consumer interface, b/c we operate on interfaces outside rabbitmq package and in tests and we need to know the name of the consumer.

**Pull request merged**